### PR TITLE
Add Microsoft Teams Notification Channel

### PR DESCRIPTION
**Pull request by Tob0t**
This PR is adding Microsoft Teams as Laravel Notification Channel

### CODE REVIEW

File: src/MicrosoftTeamsMessage.php [hunk 1]
```diff
@@ -0,0 +1,378 @@
...
+     * @param $webhookUrl - url of webhook
+     *
+     * @throws CouldNotSendNotification
+     *
+     * @return MicrosoftTeamsMessage $this
+     */
+    public function to(?string $webhookUrl): self
+    {
+        if (! $webhookUrl) {
+            throw CouldNotSendNotification::microsoftTeamsWebhookUrlMissing();
```
Comment (atymic) [inline src/MicrosoftTeamsMessage.php:307]: Not sure why this takes a nullable string, then throws an exception?

Comment (Tob0t) [reply src/MicrosoftTeamsMessage.php:307]: Yeah I already faced that problem and changed it to a nullable string due to the following case:

I you forget to define the `to()` method or in my case the config or env setting in the to method (e.g. `->to(config('services.teams.subscriptions_url'))` than you get the following error:
```
Argument 1 passed to NotificationChannels\MicrosoftTeams\MicrosoftTeamsMessage::to() must be of the type string, null given
```
Which is not a very meaningful error imho if you don't know the architecture of the package. So I made it nullable and throw a more meaningful Exception if the param in to() is empty:
```
(NotificationChannels\\MicrosoftTeams\\Exceptions\\CouldNotSendNotification(code: 0):
Microsoft Teams webhook url is missing. Please add it as param over the MicrosoftTeamsMessage::to($url) method or return it in the notifiable model by providing the method Model::routeNotificationForMicrosoftTeams().
```

I can change it back, but this was just my thinking. What do you think, maybe there is a better approach?

Comment (atymic) [reply src/MicrosoftTeamsMessage.php:307]: Ahh, I understand (I didn't see that you are calling this method with the result of the routing on the notifiable model)
All good, thanks for the explaination!

**Comment by atymic** [comment]
Merge when ready :)

**Pull request merged**